nwb_conversion_tools/utils/json_schema.py

```diff
@@ -33,7 +33,7 @@ def append_replace_dict_in_list(d, ls, k):
         # Index where the value dictionary[k] exists in the list of dicts
         ind = np.where([d[k] == i[k] for i in ls])[0]
         if len(ind) > 0:
-            ls[ind[0]] = d
+            ls[ind[0]].update(d)
```

Comment:
should this be a recursive call to `deep_dict_update`?

Reply:
For maximum safety, I would think yes - for cases where that dictionary itself has keys that contain multiple hierarchies

Reply:
agreed, I have made changes in the latest commits